.github/workflows/task-coverage-bootstrapper.yml

```diff
@@ -0,0 +1,129 @@
+        required: true
+        type: string
+      bootstrapper-artifacts-hash:
+        description: "Hash used to retrieve the artifact"
+        required: true
+        type: string
+    secrets:
+      ETH_FORK_URL:
+        required: true
+      GITHUB_GATEWAY_KEY:
+        required: false
+
+jobs:
+  test-with-coverage-bootstrapper:
+    permissions:
+      pull-requests: write
+    runs-on: karnot-arc-runner-set
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      # Setup the environment with all necessary dependencies
+      - uses: ./.github/actions/load-env
+
+      # Setup the rust environment
+      - uses: ./.github/actions/rust-setup
+        with:
+          # Uses the dev cache key as we are using the --profile dev later
+          cache-key: COVERAGE
+          rust-version: ${{ env.BUILD_RUST_VERSION }}
+          scarb-version: ${{ env.BUILD_SCARB_VERSION }}
+          install-mold: true
+          install-scarb: true
+          install-foundry: true
+          foundry-version: stable
+          install-node: true
+          node-version: ${{ env.BUILD_NODE_VERSION }}
+          install-ganache: true
+          ganache-version: ${{ env.BUILD_GANACHE_VERSION }}
+          install-solc: true
+          solc-version: ${{ env.BUILD_SOLC_VERSION }}
+          python-version: ${{ env.BUILD_PYTHON_VERSION }}
+          build-snos: true
+
+      - name: Install cargo-llvm-cov
+        uses: taiki-e/install-action@cargo-llvm-cov
+      - name: Install nextest
+        uses: taiki-e/install-action@nextest
+      - name: Download Madara binary
+        uses: actions/download-artifact@v4
+        with:
+          name: madara-binary-${{ inputs.madara-binary-hash }}
+          path: target/release/
+      - name: Make Madara binary executable
+        run: chmod +x target/release/madara
+
+      - name: Prepare Madara Environment (Bootstrapper)
+        run: |
+          mv target/release/madara ./madara-binary          
+          echo "MADARA_BOOTSTRAPPER_MADARA_BINARY_PATH=$(pwd)/madara-binary" >> $GITHUB_ENV
+          echo "Binary path: $(pwd)/madara-binary"
+
+          # Add environment variable for devnet.yaml file path
+          echo "MADARA_BOOTSTRAPPER_CONFIG_PATH=$(pwd)/configs/presets/devnet.yaml" >> $GITHUB_ENV
+          echo "Config path: $(pwd)/configs/presets/devnet.yaml"
+
+          # Verify the config file exists
+          if [ -f "$(pwd)/configs/presets/devnet.yaml" ]; then
+            echo "Config file exists at specified path"
+          else
+            echo "Config file NOT found at specified path!"
+            ls -la $(pwd)/configs/presets/ || echo "Directory not found!"
+          fi
+      - name: Download Bootstrapper artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: bootstrapper-artifacts-${{ inputs.bootstrapper-artifacts-hash }}
+          path: artifacts
+
+      - name: Start Anvil with fork
```

Comment:
The final step `Kill Anvil` (for resources cleanup) is missing. You can copy it from other actions,